File: SDKs/googleapis.com/discoveryengine/v1alpha/java/lib/src/main/java/openapisdk/models/shared/GoogleCloudDiscoveryengineV1alphaImportUserEventsRequestInlineSource.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GoogleCloudDiscoveryengineV1alphaImportUserEventsRequestInlineSource
 * The inline source for the input config for ImportUserEvents method.
**/
public class GoogleCloudDiscoveryengineV1alphaImportUserEventsRequestInlineSource {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("userEvents")
    public GoogleCloudDiscoveryengineV1alphaUserEvent[] userEvents;
    public GoogleCloudDiscoveryengineV1alphaImportUserEventsRequestInlineSource withUserEvents(GoogleCloudDiscoveryengineV1alphaUserEvent[] userEvents) {
        this.userEvents = userEvents;
        return this;
    }
}

File: SDKs/googleapis.com/discoveryengine/v1alpha/python/src/sdk/models/shared/googleclouddiscoveryenginev1alphaimportusereventsrequestinlinesource.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import googleclouddiscoveryenginev1alphauserevent as shared_googleclouddiscoveryenginev1alphauserevent


@dataclass_json
@dataclasses.dataclass
class GoogleCloudDiscoveryengineV1alphaImportUserEventsRequestInlineSource:
    r"""GoogleCloudDiscoveryengineV1alphaImportUserEventsRequestInlineSource
    The inline source for the input config for ImportUserEvents method.
    """
    
    user_events: Optional[list[shared_googleclouddiscoveryenginev1alphauserevent.GoogleCloudDiscoveryengineV1alphaUserEvent]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('userEvents') }})
    
